modules/key-distributor/README.md

```diff
@@ -0,0 +1,166 @@
+# Service Account Key Distributor
+
+Downloading Google Cloud Service Account keys is [typically a bad idea](https://medium.com/@jryancanty/stop-downloading-google-cloud-service-account-keys-1811d44a97d9) but sometimes you can't help it.
+For example if you need to create a key to distribute to a CICD system on-prem like Jenkins, or you need to give some third party
+access to Google Cloud without giving them a user account. If you download the key and directly distribute it, not only is it
+directly visible to you, but it's in plain text in whatever storage medium you put it in (i.e. GCS, etc). A better idea would
+be to use [GPG](https://gnupg.org/) to download the key and encrypt it such that only the intended party can decrypt it. However,
+in this case, the person downloading the key still has access to see it, which means they could accidentally commit it to Github,
+send it off to a friend, etc.
+
+This solution allows operators to generate a Google Cloud service account key for another system (e.g. CICD pipeline) or user without
+ever needing to see the key itself, adhering to the principle of least privilege. This is possible using asymmetric encryption
+and the GPG software utility. This assumes the receiving service has a GPG private key which can be used to decrypt the service
+account key, and the deployer of this function has access to the GPG public key.
+
+## Architecture
+
+![](./arch.png)
+
+In this diagram, you can see the Cloud Function is used to broker the Service Account key generation and public key encryption. The operator then only needs to be granted
+access to call the function, but not to create the service account key. The function returns the encrypted key, which the operator can then distribute to
+the application or user that owns the private key.
+
+## Deployment
+```
+module "distributor" {
+  source  = "terraform-google-modules/service-accounts/google//key-distributor"
+
+  project_id      = "<project-id>"
+  org_id          = "<org-id>"
+  public_key_file = "/path/to/pubkey.asc"
```

Comment:
This will require updating the function each time you want to share with a new user (with a new public key). Is there a reason we wouldn't simply accept the desired public key as an input to function execution? I'm trying to understand the threat model here and the interaction between the module applier, the function invoker, and the eventual key user.

Reply:
If the user had control over the public key they could simply encrypt the key to themselves and view the Service Account key themselves. The goal is to separate the duties of service account key generation and distribution

Reply:
I see, so we're assuming the person executing the Cloud Function call is different from the person deploying the Terraform module?

Reply:
Yes that’s correct